src/miner.cpp

```diff
@@ -439,6 +465,9 @@ void BlockAssembler::addPackageTxs()
             continue;
         }
 
+        // This transaction will make it in; reset the failed counter.
+        nConsecutiveFailed = 0;
```

Comment:
Could you add more of a comment on why resetting is correct behavior here?

It seems to me at first glance, that if we fail to add something earlier, we should continue to tally those as failures?
Perhaps:
```
Assuming that below a certain threshold `T` of probability
(i.e.,`T = P(success at M | failure at M-1....0)` where `M = 1000`) 
of adding something to the block we want to give up,
 we expect that 
`1000>D>0. P(success at N+D+1 | failure at N+D,... success at N, failure at N-1, ...) 
> P(success at N+D | failure at N, failure at N-1, ...)`?
```

Maybe not the most accurate, but would be helpful for future reviewers trying to understand what the intention is.